src/main/java/io/vertx/core/net/TrafficShapingOptions.java

```diff
@@ -127,7 +155,7 @@ public TrafficShapingOptions setCheckIntervalForStats(long checkIntervalForStats
    * @return a reference to this, so the API can be used fluently
    */
   public TrafficShapingOptions setCheckIntervalForStatsTimeUnit(TimeUnit checkIntervalForStatsTimeUnit) {
-    this.maxDelayToWaitTimeUnit = checkIntervalForStatsTimeUnit;
```

Comment:
this assignment is incorrect, and is causing NPEs when the check interval is set